lua/completion-treesitter/source.lua

```diff
@@ -0,0 +1,91 @@
+local ts_locals = require'nvim-treesitter.locals'
+local parsers = require'nvim-treesitter.parsers'
+local ts_utils = require'nvim-treesitter.ts_utils'
+
+local M = {}
+
+M.match_kinds = {
+  var = 'v',
+  method = 'm',
+  ['function'] = 'f'
+}
+
+local function get_smallest_context(source)
+  -- Step 1 get current context
+  local scopes = ts_locals.get_scopes()
+  local current = source
+
+  while current ~= nil and not vim.tbl_contains(scopes, current) do
+    current = current:parent()
+  end
+
+  return current or nil
+end
+
+local function prepare_match(match, kind)
+  local matches = {}
+
+  if match.node then
+    table.insert(matches, { kind = M.match_kinds[kind] or '', full = match.node, def = match.node })
```

Comment:
I am not sure we need both `full` and `def` as they now are the same

Reply:
I thought the same thing. I was going to remove it, but forgot.